docs-src/security-model.md

```diff
@@ -72,14 +72,14 @@ readable by most roles, so instead we'll focus on write access for this model.
 The extra configuration options are not possible to control with RBAC. Instead,
 they will be controlled with configuration fields on GatewayClasses:
 
-* **allowedGatewayNamespaceSelector**: This field is a selector of namespaces
-  that Gateways can use this GatewayClass from. This is a standard Kubernetes
-  LabelSelector, a label query over a set of resources. The result of
-  matchLabels and matchExpressions are ANDed. Controllers must not support
-  Gateways in namespaces outside this selector. An empty selector (default)
-  indicates that Gateways can use this GatewayClass from any namespace. This
-  field is intentionally not a pointer because the nil behavior (no namespaces)
-  is undesirable here.
+* **allowedGatewayNamespaces**: AllowedGatewayNamespaces is a selector of
+  namespaces that Gateways of this class can be created in. This selector may be
+  changed in a way that invalidates existing Gateways. Implementations may
+  choose to immediately terminate related infrastructure. If implementations can
+  differentiate between provisioned and pending Gateways, they may choose to
+  continue to support any Gateways that were provisioned before this field
+  changed. To ensure this field is not bypassed, this differentiation should not
+  be based on the Gateway status or other fields that users can manipulate.
```

Comment:
It would help to state on what this differentiation _should_ be based, namely the _`GatewayClass`_ status (per the godoc for `AllowedGatewayNamespaces`: "a `Gateway` must only be considered provisioned if it appears in the `ProvisionedGateways` list in `GatewayClassStatus`").

Reply:
Thanks for catching this, updated this to match the godocs.